src/main/scala/sbtprotoc/ProtocPlugin.scala

```diff
@@ -309,6 +311,56 @@ object ProtocPlugin extends AutoPlugin with Compat {
       }
     }
 
+  // Forward lines from an OutputStream to a given logger at a specified level
+  private class LoggingOutputStream(logger: Logger, level: Level.Value) extends OutputStream {
+    private val baos = new ByteArrayOutputStream
+    override def write(b: Int): Unit = {
+      baos.write(b)
+      val lineSoFar = baos.toString
+      if (lineSoFar.endsWith(System.lineSeparator)) {
+        logger.log(level, lineSoFar.stripSuffix(System.lineSeparator))
+        baos.reset
+      }
+    }
+  }
+
+  // Wrap an OutputStream, allowing to intercept calls locally (on a per thread basis)
+  private class ThreadLocalOutputStream(default: OutputStream) extends OutputStream {
+    private val threadLocal = new ThreadLocal[OutputStream] {
+      override def initialValue() = default
+    }
+
+    def withValue[T](outputStream: OutputStream)(thunk: => T): T =
+      try {
+        threadLocal.set(outputStream)
+        thunk
+      } finally {
+        threadLocal.remove()
+      }
+
+    override def write(i: Int): Unit = threadLocal.get.write(i)
+    override def flush(): Unit       = threadLocal.get.flush()
+    override def close(): Unit       = threadLocal.get.close()
+  }
+
+  // Hijack System.out so that we can intercept debug println from protoc-jar
+  private[this] val stdoutDispatcher = new ThreadLocalOutputStream(System.out)
```

Comment:
Maybe instead of this add an override in protoc-jar that takes an output stream for protoc-jar own logging?

Reply:
As mentioned in the description, that's what I initially tried: https://github.com/os72/protoc-jar/compare/master...bjaglin:logs, but I was not very satisfied as:
- the overload cascades to all methods as everything is `static public`
- for any new functionnality, it would be very easy to log to sdtout by mistake even though the calling context asked for a specific stream

Adding a proper logging framework would be an alternative, but I wasn't sure the feature was worth adding a dependency to this minimalist lib.

Happy to hear your thoughts!

Reply:
I appreciate the effort you put into this. I'm concerned about the complexity and potential problems this might introduce (especially on operating systems I don't use). I would prefer to see if we can get the changes into protoc-jar first.

Reply:
Totally understandable, I'll follow up there. Meanwhile, I have force-pushed to remove the commit that was capturing `protoc-jar`'s own output, since the core of this PR was to capture `protoc` stdout/stderranyway  (which does not need the hack, just the breaking change `SettingKey` -> `TaskKey` discussed above).

Reply:
https://github.com/os72/protoc-jar/issues/74